pep-0692.rst

```diff
@@ -0,0 +1,643 @@
+ideal, or to refactor the function to use explicit keyword arguments, which
+often exceeds the scope of time and effort allocated to adding type hinting
+and, as any code change, introduces risk for both project maintainers and
+users. In that case hinting ``**kwargs`` using a ``TypedDict`` as described in
+this PEP will not require refactoring and function body and function
+invocations could be appropriately type checked.
+
+Another useful pattern that justifies using and typing ``**kwargs`` as proposed
+is when the function's API should allow for optional keyword arguments that
+don't have default values.
+
+However, it has to be pointed out that in some cases there are better tools
+for the job than using ``TypedDict`` to type ``**kwargs`` as proposed in this
+PEP. For example, when writing new code if all the keyword arguments are
+required or have default values then writing everything explicitly is better
+than using ``**kwargs`` and a ``TypedDict``\:
+
+::
+
+    def foo(name: str, year: int): ...  # Preferred way.
+    def foo(**kwargs: **Movie): ...
+
+Similarly, when type hinting third party libraries via stubs it is again better
+to state the function signature explicitly - this is the only way to type such
+a function if it has default parameters. Another issue that may arise in this
+case when trying to type hint the function with a ``TypedDict`` is that some
+standard function arguments may be treated as keyword only:
+
+::
+
+    def foo(name, year): ...              # Function in a third party library.
+
+    def foo(**Movie): ...                 # Function signature in a stub file.
+
+    foo("Life of Brian", 1979)            # This would be now failing type
+                                          # checking but is fine.
+
+    foo(name="Life of Brian", year=1979)  # This would be the only way to call
+                                          # the function now that passes type
+                                          # checking.
+
+Therefore, in this case it is again preferred to type hint such function
+explicitly as:
+
+::
+
+    def foo(name: str, year: int): ...
+
+Grammar Changes
+===============
+
+This PEP requires a grammar change so that the double asterisk syntax is
+allowed for ``**kwargs`` annotations. The proposed change is to extend the
+``kwds`` rule as follows:
+
+Before:
+
+::
+
+    kwds: '**' param_no_default 
+
+After:
+
+::
+
+    kwds:
+        | '**' param_no_default_double_star_annotation
+        | '**' param_no_default
+
+    param_no_default_double_star_annotation:
+        | param_double_star_annotation & ')'
+
+    param_double_star_annotation: NAME double_star_annotation
+
+    double_star_annotation: ':' double_star_expression
+
+    double_star_expression: '**' expression
+
+A new AST node needs to be created so that type checkers can differentiate the
+semantics of the new syntax and take into account that the ``**kwargs`` should
```

Comment:
> differentiate the semantics of the new syntax

I didn't understand what this fragment was trying to say. Differentiate what from what else?